pkg/hub/managedclusterset/controller.go

```diff
@@ -42,121 +40,56 @@ func NewManagedClusterSetController(
 	clusterInformer clusterinformerv1.ManagedClusterInformer,
 	clusterSetInformer clusterinformerv1beta1.ManagedClusterSetInformer,
 	recorder events.Recorder) factory.Controller {
-
-	controllerName := "managed-clusterset-controller"
-	syncCtx := factory.NewSyncContext(controllerName, recorder)
-	clusterInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc: func(obj interface{}) {
-			if clusterSetName := getClusterSet(obj); len(clusterSetName) > 0 {
-				syncCtx.Queue().Add(clusterSetName)
-				return
-			}
-		},
-		UpdateFunc: func(oldObj, newObj interface{}) {
-			oldClusterSetName, newClusterSetName := getClusterSet(oldObj), getClusterSet(newObj)
-			if oldClusterSetName == newClusterSetName {
-				return
-			}
-
-			if len(oldClusterSetName) > 0 {
-				syncCtx.Queue().Add(oldClusterSetName)
-			}
-
-			if len(newClusterSetName) > 0 {
-				syncCtx.Queue().Add(newClusterSetName)
-			}
-		},
-		DeleteFunc: func(obj interface{}) {
-			var clusterSetName string
-			switch t := obj.(type) {
-			case *clusterapiv1.ManagedCluster:
-				clusterSetName = getClusterSet(obj)
-			case cache.DeletedFinalStateUnknown:
-				if _, ok := t.Obj.(metav1.Object); !ok {
-					utilruntime.HandleError(fmt.Errorf("error decoding object tombstone, invalid type"))
-					return
-				}
-				clusterSetName = getClusterSet(t.Obj)
-			default:
-				utilruntime.HandleError(fmt.Errorf("error decoding object, invalid type"))
-				return
-			}
-
-			if len(clusterSetName) > 0 {
-				syncCtx.Queue().Add(clusterSetName)
-			}
-		},
-	})
-
 	c := &managedClusterSetController{
 		clusterClient:    clusterClient,
 		clusterLister:    clusterInformer.Lister(),
 		clusterSetLister: clusterSetInformer.Lister(),
-		eventRecorder:    recorder.WithComponentSuffix(controllerName),
+		eventRecorder:    recorder.WithComponentSuffix("managed-cluster-set-controller"),
 	}
 	return factory.New().
-		WithSyncContext(syncCtx).
 		WithInformersQueueKeyFunc(func(obj runtime.Object) string {
 			accessor, _ := meta.Accessor(obj)
 			return accessor.GetName()
 		}, clusterSetInformer.Informer()).
-		WithBareInformers(clusterInformer.Informer()).
+		ResyncEvery(5*time.Second).
 		WithSync(c.sync).
 		ToController("ManagedClusterSetController", recorder)
 }
 
 func (c *managedClusterSetController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
-	clusterSetName := syncCtx.QueueKey()
-	if len(clusterSetName) == 0 {
-		return nil
-	}
-	klog.Infof("Reconciling ManagedClusterSet %s", clusterSetName)
-
-	clusterSet, err := c.clusterSetLister.Get(clusterSetName)
-	if errors.IsNotFound(err) {
-		// cluster set not found, could have been deleted, do nothing.
-		return nil
-	}
+	klog.Infof("Reconciling All ManagedClusterSet")
+	managedClusterSets, err := c.clusterSetLister.List(labels.Everything())
```

Comment:
I do not think we should resync clusterset every poll interval.

Reply:
done
changed to watch cluster and clusterset.